include/yaclib/async/detail/core.hpp

```diff
@@ -17,33 +17,29 @@ class Core : public ResultCore<Ret> {
 
  private:
   void Call() noexcept final {
-    if (Base::_state.load(std::memory_order_acquire) == Base::State::Stopped) {
+    if (Base::GetState() == BaseCore::State::HasStop) {
       Base::Cancel();
       return;
     }
-    auto ret = [&] {
-      auto* caller = static_cast<ResultCore<Arg>*>(Base::_caller.Get());
-      if constexpr (std::is_void_v<Arg>) {
-        if (!caller) {
-          return util::Result<Arg>::Default();
-        }
+    auto* caller = static_cast<ResultCore<Arg>*>(Base::_caller.Get());
+    if constexpr (std::is_void_v<Arg>) {
+      if (!caller) {
+        Base::SetResult(_functor.Wrapper(util::Result<Arg>::Default()));
+        return;
       }
-      return std::move(caller->Get());
-    }();
-    Base::_caller = nullptr;
-    Base::SetResult(_functor.Wrapper(std::move(ret)));
+    }
+    Base::SetResult(_functor.Wrapper(std::move(caller->Get())));
   }
 
-  InvokeType _functor;
-};
-
-template <typename Result>
-class Core<Result, void, void> : public ResultCore<Result> {
-  void Call() noexcept final {
-    assert(false);  // This class using only via promise
```

Comment:
- [ ]  NIT: remember to clean `#include`'s section after stopping using any declarations from some specific header.